src/lib.rs

```diff
@@ -181,6 +180,7 @@ pub use {
 
 #[cfg(feature = "mock_base")]
 pub use self::{
+    authority::Authority,
```

Comment:
Could we make `authority::Authority` pub(crate) if not mock_base so we can still have the useful names at root level?

Reply:
So I think there are two ways to look at this:
1. There is some value in having the `lib.rs` file be clean and minimal so when a newcomer looks at it, they immediately see what the public API is without it being cluttered with internal stuff
2. On the other hand, there is also value in having some commonly used items re-exported at the top level, so they are easier to use.
Neither of these is particularly strong point and I don't really lean towards either of them, which is why I'll defer this to whatever you and/or @dirvine decide.

Reply:
One thing about point 2: if an item is re-exported for mock_base, it might be good to re-export it for crate otherwise so it can be used the same way in test and in crate. But whatever David decides.

Reply:
The reason to re-export something for mock_base is mostly so that it can be used in the integration tests, not to make it convenient to use in unit tests.

Reply:
If possible I would go for clean lib.rs. However, we don't want to take it too far. So here I would be just ever so slightly in favour of this pub behind the feature flag as it is in the PR.